### ISSUE TITLE
Introduce free-form text sections for detailed usage and examples

### ISSUE THREAD

**Issue by ebeahan** [elastic/ecs#932]
**Summary**

Introduce a concept into the ECS repo which captures extended usage examples. These examples would appear in the ECS documentation to start and potentially other uses later on.

**Motivation**:

Real-world examples are powerful tools to help users better grasp ECS concepts and how users can better map their custom data sources. In addition to the field specifications, ECS also provides guidance on the preferred design patterns for modeling particular events.

A wealth of examples and past discussion is captured in the repo and on Discuss, but not in the ECS docs. Having such examples incorporated into the docs would provide a single, authoritative location. Restructuring could also allow us to include diagrams, code snippets, and other visual aids.

Issues touching on or providing such examples:

* #516
* #860 
* #809 
* #300 

**Design Proposal**:

* Refactor the existing `schemas` directory
    * As of ECS 1.5.0, the `schemas` directory contains a collection of YAML files and a README.
    * create subdirectories named after the fieldset
    * Each subdirectory contains the schema file for that fieldset. It can also contain an AsciiDoc formatted `usage` or `example` file.

* Refactor the field reference doc section

Continue to provide the table of fields within a fieldset, but include links to free-form text section that allow room for extended descriptions, code snippets, visuals, etc.

Simple (and not final) mockup:

<img width="929" alt="Screen Shot 2020-08-14 at 11 25 59 AM" src="https://user-images.githubusercontent.com/7226265/90271297-2e485200-de21-11ea-8ea8-e954ef0834ad.png">

* The AsciiDoc generator will be updated to support this refactoring in the docs

**Comment by webmat** [comment]
Super excited about this!

The screenshot you've included seems to be about the full page redesign, rather than the plain text section per field set. Not sure if that was intentional?

I very much want these two things, the free form text section and the page redesign. However I think the work could be split in two. I've been envisioning the free form text as potentially a second page per field set, rather than a new section in the same page. If you think this makes sense, it would simplify splitting the work in two. WDYT?

**Comment by ebeahan** [comment]
Yes - Agree if we separate the pages the work should be split.

I had the two tasks separated initially. I have been leaning towards the single page approach (text section + page redesign simultaneously), but after compiling some thoughts (👇 ) this may the less appealing of the two. I certainly want to avoid introducing any confusion. 😅 

---

### Thoughts

**Single page**

* Incorporate any additional text or examples for each field underneath that field's section.
* Retain a single page per field set
* I don't see  all field sets needing the second page (e.g. `ecs.*`). Always have one page containing all the details vs. some field sets having multiple pages.
* One larger effort to implement vs. two smaller ones

**Separate pages**

* Examples could be grouped by their use case vs. which field(s). I believe will be more clear for users.
* A separate page avoids any confusion over which field section an example resides under (example: we want to express the intended usage of `source.address`, and `source.domain` or `source.ip`, which field does that usage belong under under the single page design?).
* Having the second page per field set does provide more space for visuals, other tables, diagrams, code snippets etc.
* We can enjoy the plain text additions while we work the page redesign separately 😎

**Comment by webmat** [comment]
Yes the "free text" docs was always meant to be one (optional) big blob per **field set**, where we can have illustrations, talk about how fields in the field set relate to each other, and so on.

This will give us the ultimate flexibility in explaining each field set holistically, without being constrained by only having a paragraph per field available to us.

Having these as two distinct pages also has the advantage of keeping one page with the regular structure & constraints, and a companion page that is free form, but doesn't affect the structure of the other page.

**Comment by webmat** [comment]
We may want to manually sketch the nav on these optional pages, before spending too much time on code.

**Comment by ebeahan** [comment]
@webmat Sounds good :smile: . I agree we should have a rough mockup so discuss before getting into the implementation too much.

I'll edit this issue to better reflect the page redesign, and open a second issue to focus on the free text page implementation.

**Comment by webmat** [comment]
Awesome, thanks!

Perhaps also link to the public google doc in the body of this issue?

**Comment by ebeahan** [comment]
Also noting we should have a direct link to each of the fields to quickly share with users.
